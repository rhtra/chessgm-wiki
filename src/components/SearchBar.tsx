import React from "react";

type Props = {
  value: string;
  onChange: (value: string) => void;
};

export default function SearchBar({ value, onChange }: Props) {
  return (
    <div className="toolbar">
      <input
        className="input"
        placeholder="Search usernameâ€¦"
        value={value}
        onChange={(e) => onChange(e.target.value)}
      />
    </div>
  );
}
