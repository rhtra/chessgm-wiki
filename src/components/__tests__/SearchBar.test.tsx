import { render } from "@testing-library/react";
import SearchBar from "../SearchBar";

describe("SearchBar snapshots", () => {
  it("matches snapshot when empty", () => {
    const { asFragment } = render(<SearchBar value="" onChange={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with value", () => {
    const { asFragment } = render(
      <SearchBar value="magnus" onChange={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
