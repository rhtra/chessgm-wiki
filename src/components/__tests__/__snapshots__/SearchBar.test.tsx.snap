// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SearchBar snapshots matches snapshot when empty 1`] = `
<DocumentFragment>
  <div
    class="toolbar"
  >
    <input
      class="input"
      placeholder="Search username…"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`SearchBar snapshots matches snapshot with value 1`] = `
<DocumentFragment>
  <div
    class="toolbar"
  >
    <input
      class="input"
      placeholder="Search username…"
      value="magnus"
    />
  </div>
</DocumentFragment>
`;
